pkg/snapshot/layered_map.go

```diff
@@ -62,14 +63,17 @@ func (l *LayeredMap) Key() (string, error) {
 }
 
 // GetFlattenedPathsForWhiteOut returns all paths in the current FS
-func (l *LayeredMap) getFlattenedPathsForWhiteOut() map[string]struct{} {
+func (l *LayeredMap) getFlattenedPaths() map[string]struct{} {
 	paths := map[string]struct{}{}
 	for _, l := range l.layers {
 		for p := range l {
-			if strings.HasPrefix(filepath.Base(p), ".wh.") {
-				delete(paths, p)
+			basename := filepath.Base(p)
+			if strings.HasPrefix(basename, archive.WhiteoutPrefix) {
+				deletedFile := filepath.Join(filepath.Dir(p), strings.TrimPrefix(basename, archive.WhiteoutPrefix))
+				delete(paths, deletedFile)
```

Comment:
Here I delete the corresponding file, also `.wh.` prefixed files never enter the `paths` map.